apps/balancer-tools/src/app/apr/round/(components)/MultiSelectDropdown.tsx

```diff
@@ -0,0 +1,172 @@
+                            idx,
+                            selectedItems,
+                            setSelectedItems,
+                            onSelectionItemsChange,
+                          )
+                        }
+                      >
+                        <span>{value}</span>
+                        <span>
+                          <Cross1Icon />
+                        </span>
+                      </button>
+                    );
+                  })}
+
+                  <input
+                    placeholder={labelText}
+                    {...getInputProps()}
+                    onFocus={() => {
+                      toggleMenu();
+                    }}
+                    onKeyDown={(e) => {
+                      handleOKeyDown(e);
+                    }}
+                    className="w-full border-none py-2 pl-3 pr-10 text-sm leading-5 outline-none text-white bg-transparent"
+                    type="text"
+                  />
+                </div>
+              </div>
+
+              {isOpen && (
+                <div className="absolute z-50 my-2 flex max-h-52 flex-col overflow-y-scroll rounded border-[1px] border-blue6 bg-blue3 scrollbar-thin scrollbar-track-blue2 scrollbar-thumb-slate12 w-full">
+                  {items.filter(
+                    (item) =>
+                      !selectedItems.find((selected) => selected === item) &&
+                      item.includes(String(inputValue)),
+                  ).length > 0 ? (
+                    items
+                      .filter(
+                        (item) =>
+                          !selectedItems.find(
+                            (selected) => selected === item,
+                          ) && item.includes(String(inputValue)),
+                      )
+                      .map((item, idx) => {
+                        return (
+                          <div
+                            className={`flex w-full flex-col items-start cursor-pointer p-2 ${
+                              idx == highlightedIndex
+                                ? "bg-blue6"
+                                : "bg-transparent"
+                            }`}
+                            {...getItemProps({ item, key: item + idx })}
+                          >
+                            <div>
+                              <span>{item}</span>
+                            </div>
+                          </div>
+                        );
+                      })
+                  ) : (
+                    <div className="flex flex-col items-center text-center text-slate12 p-4">
+                      <span>Sorry, no token was found with the symbol</span>
+                    </div>
+                  )}
+                </div>
+              )}
+
+              <div></div>
+            </div>
+          );
+        }}
+      </Downshift>
+    </div>
+  );
+};
+
+/* Helper functions */
+function changeHandler(
```

Comment:
since you're not going to use these functions anywhere else, I'd suggest defining them within the component using `useCallback`